### PR TITLE
Publish `preview` built-ins to open-vsx.org nightly

### PR DESCRIPTION
**Pull request by marcdumais-work**
As well:
- Fix preview field from string to bool
- Adjusted the computed 'next' version we use - it looks
the version in vscode's package.json is already stepped for
the next release.
- Fixed bug in publish script where .vsix path was not provided
to ovsx

<a href="https://gitpod.io/#https://github.com/theia-ide/vscode-builtin-extensions/pull/30"></a>

### CODE REVIEW

**Comment by marcdumais-work** [comment]
I pushed the PR's commits on the master branch of my fork and it only misses the OVSX PAT to successfully publish. So I expect we'll have a successful publish upon commit and then nightly.

https://github.com/marcdumais-work/vscode-builtin-extensions/runs/707505930?check_suite_focus=true

File: .github/workflows/publish-vsx-next.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    branches:
+      - master
+env:
+  NODE_OPTIONS: --max-old-space-size=8192
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+      OVSX_TOKEN: ${{ secrets.OVSX_PAT}}
```
Comment (spoenemann) [inline .github/workflows/publish-vsx-next.yml:15]: The env variable should be named `OVSX_PAT`:
https://github.com/eclipse/openvsx/blob/592bf16e093cb9491cdd493d116c327e55f3304a/cli/src/publish.ts#L23

Comment (marcdumais-work) [reply .github/workflows/publish-vsx-next.yml]: done

**Comment by marcdumais-work** [comment]
Thanks Miro - Let's give this a try and see if we get a correct publish upon commit, and again later tonight.

**Comment by spoenemann** [comment]
The preview version is composed of the vscode version + vscode shortRevision, right? That means if the vscode master branch hasn't changed, publishing will fail because the version is already in the registry?

**Comment by marcdumais-work** [comment]
> The preview version is composed of the vscode version + vscode shortRevision, right? That means if the vscode master branch hasn't changed, publishing will fail because the version is already in the registry?

Yes, I think that's correct. The vscode repo sees quite a bit of action, so it should not happen often that there are no new commits on master, between two nightly builds.

**Comment by spoenemann** [comment]
Ok. If you'd like to avoid running a full build just to fail at publishing because the version is already in the registry, you could use this method to check the already published versions at the beginning:

https://github.com/eclipse/openvsx/blob/592bf16e093cb9491cdd493d116c327e55f3304a/cli/src/registry.ts#L57-L64

**Comment by marcdumais-work** [comment]
> Ok. If you'd like to avoid running a full build just to fail at publishing because the version is already in the registry, you could use this method to check the already published versions at the beginning:
> 
> https://github.com/eclipse/openvsx/blob/592bf16e093cb9491cdd493d116c327e55f3304a/cli/src/registry.ts#L57-L64
> 
> That might also be useful for non-preview versions.

Thanks Miro - this will be even more useful for solid releases - my plan so far was to try to build/publish every day, and let it fail between when we already have a release on open-vsx.

**Pull request merged**